### PR TITLE
Added reminder class

### PR DESCRIPTION
**Pull request by minpyaemoe**
Adding reminder class.

#34

### CODE REVIEW

**Comment by Nanosync** [comment]
same code coverage maintained, coveralls having problems

File: src/main/java/seedu/address/commons/core/Reminder.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package seedu.address.commons.core;
+
+import java.util.Date;
+
+/**
+ * Reminder class to notify user with deadlines.
+ */
+
```
Comment (Nanosync) [inline src/main/java/seedu/address/commons/core/Reminder.java]: can remove additional line break here

File: src/main/java/seedu/address/commons/core/Reminder.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+public class Reminder {
+    private String taskToDo;
```
Comment (Nanosync) [inline src/main/java/seedu/address/commons/core/Reminder.java]: please rename to `reminderName`

File: src/main/java/seedu/address/commons/core/Reminder.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    private Date deadline;
+
+    public Reminder(String taskToDo, Date deadline) {
+        this.taskToDo = taskToDo;
+        this.deadline = deadline;
+    }
+
+    public String getReminderMessage() {
```
Comment (Nanosync) [inline src/main/java/seedu/address/commons/core/Reminder.java:17]: `getReminderName()` to adhere to coding standards

File: src/test/java/seedu/address/commons/core/ReminderTest.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import org.junit.jupiter.api.Test;
+
+/**
+ * Contains unit test for Reminder.
+ */
+
+class ReminderTest {
+
+    @Test
+    public void getReminderMessage() {
```
Comment (Nanosync) [inline src/test/java/seedu/address/commons/core/ReminderTest.java:16]: change test to `isValidReminder()` and perform validation like null checks

File: src/test/java/seedu/address/commons/core/ReminderTest.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.Date;
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * Contains unit test for Reminder.
+ */
+
```
Comment (Nanosync) [inline src/test/java/seedu/address/commons/core/ReminderTest.java]: remove line break

File: src/test/java/seedu/address/commons/core/ReminderTest.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+class ReminderTest {
+
+    @Test
+    public void getReminderMessage() {
+        Reminder dummyReminder = new Reminder("Pay Bill", new Date(2019, 10, 10));
+        assertEquals("Pay Bill", dummyReminder.getReminderMessage());
+    }
+
+    @Test
+    public void getDeadline() {
```
Comment (Nanosync) [inline src/test/java/seedu/address/commons/core/ReminderTest.java:22]: change test to `isValidDeadline()` or merge it with `isValidReminder()`

**Review by Nanosync — CHANGES_REQUESTED** [review]

File: src/test/java/seedu/address/commons/core/ReminderTest.java (hunk 4 above)
Comment (choongyx) [reply src/test/java/seedu/address/commons/core/ReminderTest.java:16]: The convention for test names is `methodName_testScenario_expectedResult`.

File: src/test/java/seedu/address/commons/core/ReminderTest.java (hunk 6 above)
Comment (austinsantoso) [reply src/test/java/seedu/address/commons/core/ReminderTest.java:22]: why should it be separated? I think it should be separated as they test different things.

**Review by Nanosync — APPROVED** [review]
excellent job Min!

**Pull request merged**